src/compiler/parser.ts

```diff
@@ -1537,6 +1537,138 @@ namespace ts {
             return false;
         }
 
+        const viableKeywordSuggestions = Object.keys(textToKeywordObj).filter(keyword => keyword.length > 2);
+
+        function parseSemicolonAfter(parentKind: SyntaxKind, expression: Expression | PropertyName, initializer?: Node) {

```

Comment:
Give a little bit of motivation for this function - what's it going to try to do? When is it best used?

Reply:
It's going to try its hardest to give us good syntax errors. That's all the motivation it needs.

_(I'll see myself out...)_